src/convert/formula.jl

```diff
@@ -29,18 +29,16 @@ function rcopy(::Type{Expr}, l::Ptr{LangSxp})
         end
         f = Expr(:call, op, cargs...)
     end
-    # unwind these opeators
+    # unwind these operators
     if op in (:+, :*, :&) && isa(f.args[2], Expr) && f.args[2].args[1] == op
         f = Expr(:call, op, f.args[2].args[2:end]..., f.args[3])
     end
     f
 end
 
-function rcopy(::Type{Formula}, l::Ptr{LangSxp})
-    ex_orig = rcopy(Expr, l)
-    ex = parse!(copy(ex_orig))
-    lhs, rhs = ex.args[2:3]
-    Formula(ex_orig, ex, lhs, rhs)
+function rcopy(::Type{FormulaTerm}, l::Ptr{LangSxp})
+    expr = rcopy(Expr, l)
+    @eval StatsModels.@formula($expr)
```

Comment:
We shouldn't eval inside a function. Is there another interface we can use?

Reply:
I'm afraid not.  At least not if we want to support calls to functions that are not part of the formula language (like `log`).

Reply:
the problem is that eval is called at top-level, so this probably won't actually work.

Reply:
I would be tempted to merge this PR and get a new version of RCall on the registry, even if it is likely that this will not work perfectly.  Right now you can't install RCall if you want to have recent versions of GLM, MixedModels, etc. installed.

Reply:
I think it does work to call `@eval` in this case.  At least the following simple test seems okay.
```julia
julia> rcopy(R"y ~ 1 + A + B + log(C)")
FormulaTerm
Response:
  y(unknown)
Predictors:
  1
  A(unknown)
  B(unknown)
  (C)->log(C)
```

Reply:
The question is if it works correctly from inside a function?

Reply:
How deeply inside a function do you need?

```
julia> using RCall

R> form <- y ~ 1 + A + B + log(C)

julia> function foo()
           rcopy(R"form")
       end
foo (generic function with 1 method)

julia> foo()
FormulaTerm
Response:
  y(unknown)
Predictors:
  1
  A(unknown)
  B(unknown)
  (C)->log(C)
```